### PR TITLE
Use change_of_subject gem to transform archaic subject terms

### LINKED ISSUE #1088: create an LCSH masking mechanism as in "Change the Subject"
To mask offensive Library of Congress Subject Headings, this [remapping list](https://github.com/pulibrary/bibdata/blob/6630d8cadbfb8acd8a34aff7664e8821acb313d6/marc_to_solr/lib/change_the_subject/change_the_subject.yml) is in place in the library's catalog to hide offensive LCSH from users. Catalogers and processing archivists continue to need to use the LCSH to link our resources to others through that widely-used controlled vocab.

The Inclusive and Reparative Metadata Working Group is responsible for creating the list of LCSH terms to replace, along with the replacement terms PUL will use to cover LCSH terms. For instance, IRMWG has recently approved changing the terms "Indians of North America", "Indians of South America", and "Indians of Mexico" (all used in this [finding aid](https://findingaids.princeton.edu/catalog/WC064#find-more), as one example). We request that those terms be replaced with "Indigenous peoples of North America," etc., in the PULFA user interface. Changes in PULFA should align with changes in the PUL catalog, maintained on github in the list linked above (Kevin Reiss, Carolyn Cole, Christina Chortaria, and Jane Sandberg, maybe now also Max Kadel and Ryan Laddusaw have been involved in this work). In finding aids, LCSH are used at the collection level and also at the component level. Please let us know if the type of masking requested can be based on text strings, or if we would need to match up SH records in ASpace (yikes!) with the replacement terms. 

## Success Criteria

As an example, if I search for "Indians of Mexico", and click the component which matches, I see it say "Indigenous peoples of Mexico".

### PR DESCRIPTION
**Pull request by eliotjordan**
Closes #1088

### CODE REVIEW

**Comment by eliotjordan** [comment]
Related: https://github.com/pulibrary/pul_solr/pull/350

File: .circleci/config.yml [hunk 1]
```diff
@@ -31,7 +31,7 @@ jobs:
           name: Update bundler
           command: |
             echo 'export BUNDLER_VERSION=2.2.16' >> $BASH_ENV
-            gem install bundler -v 2.2.16
+            gem install bundler -v 2.3.18
```
Comment (hackartisan) [inline .circleci/config.yml:34]: oh good catch this is what we already have in Gemfile.

File: lib/pulfalight/traject/ead2_component_config.rb [hunk 2]
```diff
@@ -457,14 +457,30 @@
 to_field "geogname_ssm", extract_xpath("./controlaccess/geogname")
 to_field "places_ssim", extract_xpath("./controlaccess/geogname")
 
-to_field "access_subjects_ssim", extract_xpath("./controlaccess", to_text: false) do |_record, accumulator|
-  accumulator.map! do |element|
+to_field "access_subjects_ssim" do |record, accumulator|
```
Comment (hackartisan) [inline lib/pulfalight/traject/ead2_component_config.rb:460]: We got some more info on these fields? Curious what you found out.

File: lib/pulfalight/traject/ead2_component_config.rb [hunk 3]
```diff
@@ -616,8 +632,21 @@
...
+end
+
+# For search only
+to_field "archaic_subject_terms_ssim" do |record, accumulator, context|
+  values = record.xpath('./controlaccess/subject[not(@source="local")]').map(&:text)
+  values = values.map(&:strip)
+  processed_values = context.output_hash.fetch("subject_terms_ssim", [])
+
+  # Return array of original archaic values
+  accumulator.concat(values - processed_values)
```
Comment (hackartisan) [inline lib/pulfalight/traject/ead2_component_config.rb:649]: 👍🏻 it only includes the archaic values

**Review by hackartisan — CHANGES_REQUESTED** [review]
🎉 looks awesome. Blocking on my question but I don't have any requested changes.

Should we reindex staging to take a look before we merge and deploy to prod?

File: lib/pulfalight/traject/ead2_component_config.rb (hunk 2 above)
Comment (eliotjordan) [reply lib/pulfalight/traject/ead2_component_config.rb:460]: They are used in the generic text search as well as returned hit highlighting.

**Comment by eliotjordan** [comment]
@hackartisan Last time I tried reindexing the staging site it failed. `Pulfalight::MissingRepositoryError`. Perhaps we should wipe the index and start from scratch?

**Review by tpendragon — APPROVED** [review]

File: lib/pulfalight/traject/ead2_component_config.rb (hunk 2 above)
Comment (hackartisan) [reply lib/pulfalight/traject/ead2_component_config.rb:460]: And they'll include values we want to replace?

**Review by hackartisan — APPROVED** [review]

**Pull request merged**